tools/refguide_check.py

```diff
@@ -790,37 +792,27 @@ def _run_doctests(tests, full_name, verbose, doctest_warnings):
     runner = DTRunner(full_name, checker=Checker(), optionflags=flags,
                       verbose=verbose)
 
-    output = []
+    output = io.StringIO(newline='')
     success = True
-    def out(msg):
-        output.append(msg)
-
-    class MyStderr:
-        """
-        Redirect stderr to the current stdout
-        """
-        def write(self, msg):
-            if doctest_warnings:
-                sys.stdout.write(msg)
-            else:
-                out(msg)
 
-        # a flush method is required when a doctest uses multiprocessing
-        # multiprocessing/popen_fork.py flushes sys.stderr
-        def flush(self):
-            if doctest_warnings:
-                sys.stdout.flush()
+    # Redirect stderr to the stdout or output
+    tmp_stderr = sys.stdout if doctest_warnings else output
+
+    @contextmanager
+    def temp_cwd():
+        cwd = os.getcwd()
+        try:
+            tmpdir = tempfile.mkdtemp()
```

Comment:
This line should be outside the try finally, like it was before.

Otherwise if it falls you'll UnboundLocalError.

Reply:
Done.